dapp/tests/unit/validation.test.js

```diff
@@ -0,0 +1,183 @@
+const {
+  validateWorkerEnv,
+  validateAppSecret,
+  validateRequesterSecret,
+  validateProtectedData,
+} = require('../../src/validation');
+
+describe('validateWorkerEnv function', () => {
+  let testedObj;
+
+  beforeEach(() => {
+    testedObj = {
+      IEXEC_OUT: '/path/to/iexec/out',
+    };
+  });
+
+  it('should not throw an error if all input values are valid', () => {
+    expect(() => validateWorkerEnv(testedObj)).not.toThrow();
+  });
+
+  it('should throw an error if any required input is missing', () => {
+    delete testedObj.IEXEC_OUT;
+    expect(() => validateWorkerEnv(testedObj)).toThrow(
+      Error('Worker environment error: "IEXEC_OUT" is required')
+    );
+  });
+});
+
+describe('validateAppSecret function', () => {
+  let testedObj;
+
+  beforeEach(() => {
+    testedObj = {
+      MJ_APIKEY_PUBLIC: 'api_public_key',
+      MJ_APIKEY_PRIVATE: 'api_private_key',
+      MJ_SENDER: 'sender@example.com',
+    };
+  });
+
+  it('should not throw an error if all input values are valid', () => {
+    expect(() => validateAppSecret(testedObj)).not.toThrow();
+  });
+
+  it('should throw an error if any required input is missing', () => {
+    const { MJ_APIKEY_PUBLIC, ...missingMJAPIKEYPUBLIC } = testedObj;
+    expect(() => validateAppSecret(missingMJAPIKEYPUBLIC)).toThrow(
+      /"MJ_APIKEY_PUBLIC" is required/i
+    );
+    const { MJ_APIKEY_PRIVATE, ...missingMJAPIKEYPRIVATE } = testedObj;
+    expect(() => validateAppSecret(missingMJAPIKEYPRIVATE)).toThrow(
+      /"MJ_APIKEY_PRIVATE" is required/i
+    );
+    const { MJ_SENDER, ...missingMJSENDER } = testedObj;
+    expect(() => validateAppSecret(missingMJSENDER)).toThrow(
+      /"MJ_SENDER" is required/i
+    );
+  });
```

Comment:
Had a hard time reading the spread operator from a de-structured object.

I would not bother re-using the beforeEach created object:

```suggestion
  it('should throw an error if any required input is missing', () => {
    expect(() =>
      validateAppSecret({
        MJ_APIKEY_PRIVATE: 'api_private_key',
        MJ_SENDER: 'sender@example.com',
      })
    ).toThrow(/"MJ_APIKEY_PUBLIC" is required/i);

    expect(() =>
      validateAppSecret({
        MJ_APIKEY_PUBLIC: 'api_public_key',
        MJ_SENDER: 'sender@example.com',
      })
    ).toThrow(/"MJ_APIKEY_PRIVATE" is required/i);

    expect(() =>
      validateAppSecret({
        MJ_APIKEY_PUBLIC: 'api_public_key',
        MJ_APIKEY_PRIVATE: 'api_private_key',
      })
    ).toThrow(/"MJ_SENDER" is required/i);
  });
```